### PR TITLE
[ci]: DOPS-1789: Update iroha2 Dockerfiles & Workflows

### PR DESCRIPTION
**Pull request by BAStos525**
Signed-off-by: Vasilii Zyabkin [vzyabkin@soramitsu.co.jp](mailto:vzyabkin@soramitsu.co.jp)

### Description of the Change
1. Update iroha2 Dockerfiles for ci and switch to musl architecture for building processes.
2. Update jobs for codecov coverage reports generation.

### Issue
1. We need to update iroha2 ci process in order to improve it, make faster and get rid of some bottlenecks.
2. Codecov coverage reports takes so much time, so we have to try to reuse the same reports in the case if repository state is the same.

### Benefits
1. Reduce the size of images, switch to alpine and Arch base images, increase the speed of building process.
3. Implement the process of iroha-2 CI the reuse of codecov reports for the same state of the repository for which GH Actions workflow is running.

### Possible Drawbacks
Some possibility of failing compilation on any building stages is possible.

### CODE REVIEW

File: rust-toolchain.toml [hunk 1]
```diff
@@ -1,5 +0,0 @@
-[toolchain]
-channel = "1.60"
```
Comment (mversic) [inline rust-toolchain.toml:2]: why remove this file?

Comment (BAStos525) [reply rust-toolchain.toml:2]: actually this file was removed by [appetrosyan](https://github.com/appetrosyan)

Comment (appetrosyan) [reply rust-toolchain.toml:2]: It gets in the way.

Comment (appetrosyan) [reply rust-toolchain.toml:2]: The problem was that it checks the toolchain updates **every time** we run any `cargo` command.

Comment (appetrosyan) [reply rust-toolchain.toml:2]: @Arjentix You introduced the file originally, do you think we can make this optional, like we do with e.g. `hooks`?

**Pull request closed without merging**